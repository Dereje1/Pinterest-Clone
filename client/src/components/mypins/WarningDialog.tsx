import React from 'react';
import Dialog from '@mui/material/Dialog';
import DialogActions from '@mui/material/DialogActions';
import DialogTitle from '@mui/material/DialogTitle';
import Button from '@mui/material/Button';

interface WarningDialogProps {
    showDialog: boolean,
    title: string,
    handleCancel: () => void,
    handleContinue: () => void
}

function WarningDialog({
  showDialog, title, handleCancel, handleContinue,
}: WarningDialogProps) {
  return (
    <Dialog
      open={showDialog}
      aria-labelledby="alert-dialog-title"
      aria-describedby="alert-dialog-description"
    >
      <DialogTitle id="alert-dialog-title">
        {title}
      </DialogTitle>
      <DialogActions>
        <Button
          id="cancel-alert"
          onClick={handleCancel}
        >
          Cancel
        </Button>
        <Button
          id="resume-alert"
          onClick={handleContinue}
          autoFocus
        >
          Continue
        </Button>
      </DialogActions>
    </Dialog>
  );
}

export default WarningDialog;
