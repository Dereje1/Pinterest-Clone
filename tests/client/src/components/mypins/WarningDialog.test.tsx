import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import WarningDialog from '../../../../../client/src/components/mypins/WarningDialog';

describe('The Warning Dialog component', () => {
  let props: React.ComponentProps<typeof WarningDialog>;
  beforeEach(() => {
    props = {
      showDialog: true,
      title: 'warning title',
      handleCancel: jest.fn(),
      handleContinue: jest.fn(),
    };
  });

  test('will render...', () => {
    const wrapper = shallow(<WarningDialog {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
