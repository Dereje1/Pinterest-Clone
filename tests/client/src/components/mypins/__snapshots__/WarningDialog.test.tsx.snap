// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Warning Dialog component will render... 1`] = `
<ForwardRef(Dialog)
  aria-describedby="alert-dialog-description"
  aria-labelledby="alert-dialog-title"
  open={true}
>
  <ForwardRef(DialogTitle)
    id="alert-dialog-title"
  >
    warning title
  </ForwardRef(DialogTitle)>
  <ForwardRef(DialogActions)>
    <ForwardRef(Button)
      id="cancel-alert"
      onClick={[MockFunction]}
    >
      Cancel
    </ForwardRef(Button)>
    <ForwardRef(Button)
      autoFocus={true}
      id="resume-alert"
      onClick={[MockFunction]}
    >
      Continue
    </ForwardRef(Button)>
  </ForwardRef(DialogActions)>
</ForwardRef(Dialog)>
`;
